File: tv/src/main/java/com/synergics/stb/iptv/leanback/BaseApplication.java
package com.synergics.stb.iptv.leanback;

import android.app.Application;
import android.arch.persistence.room.Room;
import android.content.SharedPreferences;

import com.synergics.stb.iptv.leanback.models.db.MyDatabase;

public class BaseApplication extends Application {
    public static final String DBNAME = "stb-iptv";
    public static BaseApplication instance;
    public static SharedPreferences sp;
    public static MyDatabase database;

    @Override
    public void onCreate() {
        super.onCreate();
        sp = getSharedPreferences("pengaturan", MODE_PRIVATE);
        database = Room.databaseBuilder(this, MyDatabase.class, DBNAME).allowMainThreadQueries().fallbackToDestructiveMigration().build();
        instance = this;
    }
}

File: tv/src/main/java/com/synergics/stb/iptv/leanback/BaseApplication.kt
package com.synergics.stb.iptv.leanback

import android.app.Application
import android.content.SharedPreferences
import androidx.room.Room
import com.synergics.stb.iptv.leanback.models.db.MyDatabase

class BaseApplication : Application() {
    override fun onCreate() {
        super.onCreate()
        sp = getSharedPreferences("pengaturan", MODE_PRIVATE)
        database =
            Room.databaseBuilder(this, MyDatabase::class.java, DBNAME).allowMainThreadQueries()
                .fallbackToDestructiveMigration().build()
        instance = this
    }

    companion object {
        const val DBNAME = "stb-iptv"
        var instance: BaseApplication? = null
        var sp: SharedPreferences? = null
        var database: MyDatabase? = null
    }
}